src/App.tsx

```diff
@@ -1,18 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const totalItems = items.length;
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const firstItemNum = (currentPage - 1) * perPage + 1;
+  const lastItemNum = (currentPage * perPage) > totalItems
+    ? totalItems
+    : currentPage * perPage;
+
+  const itemsToShow = items.filter((item, index) => {
+    if (index >= firstItemNum - 1 && index <= lastItemNum - 1) {
+      return item;
+    }
+
+    return 0;
```

Comment:
Callback of the filter method should return a boolean value (true or false).
Let's make it clear